Assets/Scripts/Entity/Player/InputController.cs

```diff
@@ -2,19 +2,31 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+[RequireComponent(typeof(Player))]
 [RequireComponent(typeof(PlayerMovement))]
+[RequireComponent(typeof(ShootController))]
 public class InputController : MonoBehaviour
 {
+    private Player p;
```

Comment:
I would use more descriptive names. I know it's faster to develop if you use shorter variable names, but `p` isn't very informative :D

Reply:
Fair enough.